ls1intum/Artemis
# `Programming exercises`: Handle missing test feedback as negative result for available hint selection

### Checklist
#### General
#### Server
#### Changes affecting Programming Exercises

### Motivation and Context
Fixes #5392. Missing test feedback (e.g, for skipped tests) is not handled properly and throws a `NullPointerException` when calculating the available hints for this exercise. This occurs because `Feedback::isPositive` returns a value of type `Boolean` (non-primitive). We only considered the primitive `boolean`.

### Description
With these changes, a test with missing feedback is handled as a negative result.

### Steps for Testing
Testing locally might be best, because of log and DB access. But can also be tested on a TS.

1. Create a new Programming exercise (e.g. Python)
2. Participate as a student
3. Add `import not_existing` to the start of any student's submission file. This causes the test cases to be skipped and thus have no feedback result.
4. Verify that the server logs do not contain any `NullPointerExceptions` and that the client does not receive an `InternalServerError` on the participation page.

### Review Progress
#### Code Review
#### Manual Tests

### Test Coverage
| Class/File | Branch | Line |
|------------|-------:|-----:|
| ExerciseHintService.java | 82% | 96% |

### Screenshots
Screenshot from my local DB with missing test feedback, but no errors occurring:

id, detail_text, text, result_id, type, positive
<img width="809" alt="Bildschirmfoto 2022-07-15 um 14 07 01" src="https://user-images.githubusercontent.com/73833780/179219822-bffd6d21-50ec-4dfc-aff5-9f96807b0a31.png">
